### PR TITLE
roachtest: add cluster settings operations

### PR DESCRIPTION
**Pull request by herkolategan**
This change adds operations to mutate cluster settings. The values are mutated
based on a preset frequency and RNG. For example, if the frequency of a cluster
setting operation is set to "30 minutes" it will only change to a new value
every 30 minutes, even if the operation has been run multiple times within the
same 30-minute window. Running in the same window will just set the same value
again.

Epic: None
Release Note: None

### CODE REVIEW

File: pkg/cmd/roachtest/roachtestutil/operations/dependency.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ func CheckDependencies(
 	for _, dep := range spec.Dependencies {
 		switch dep {
 		case registry.OperationRequiresNodes:
-			if len(c.Nodes()) == 0 {
```
Comment (srosenberg) [inline pkg/cmd/roachtest/roachtestutil/operations/dependency.go:32]: It's counter-intuitive that `c.Nodes()` denotes the empty set (of nodes). We should probably update the API; my quick audit of all the callers didn't yield any other than the above.

Comment (herkolategan) [reply pkg/cmd/roachtest/roachtestutil/operations/dependency.go:32]: Nice spot, I also ran into it, second commit on this PR fixes that issue 😉

Comment (herkolategan) [reply pkg/cmd/roachtest/roachtestutil/operations/dependency.go:32]: Yep, took me a few to realise why `OperationRequiresNodes` wasn't working.

File: pkg/cmd/roachtest/operations/cluster_settings.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+func timeBasedValue(
+	timeSupplier func() time.Time, frequency time.Duration, valueForSegment func(int64) string,
+) func() string {
+	return func() string {
+		segment := timeSupplier().Unix() / int64(frequency.Seconds())
+		return valueForSegment(segment)
+	}
+}
+
+// timeBasedValues returns a function that returns a value from the given list
```
Comment (itsbilal) [inline pkg/cmd/roachtest/operations/cluster_settings.go:45]: nice!

File: pkg/cmd/roachtest/operations/cluster_settings.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+	timeSupplier := func() time.Time {
+		return timeutil.Now()
+	}
+	ops := []clusterSettingOp{
+		// Converts all leases to expiration. Tradeoff between lower throughput and higher availability.
+		// Weekly cycle.
+		{
+			Name:      "kv.expiration_leases_only.enabled",
+			Generator: timeBasedValues(timeSupplier, []string{"true", "false"}, 24*7*time.Minute),
+			Owner:     registry.OwnerTestEng,
```
Comment (itsbilal) [inline pkg/cmd/roachtest/operations/cluster_settings.go:92]: Maybe the owner for this should be KV?

File: pkg/cmd/roachtest/operations/cluster_settings.go [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+		// When running multi-store with `--wal-failover=among-stores`, this configures
+		// the threshold to trigger a fail-over to a secondary store’s WAL.
+		// 20-minute cycle.
+		{
+			Name: "storage.wal_failover.unhealthy_op_threshold",
+			Generator: timeBasedRandomValue(timeSupplier, 20*time.Minute, func(rng *rand.Rand) string {
+				return fmt.Sprintf("%d", rng.Intn(246)+5)
+			}),
+			Owner: registry.OwnerTestEng,
+		},
```
Comment (itsbilal) [inline pkg/cmd/roachtest/operations/cluster_settings.go:103]: Maybe we should add one for `kv.snapshot_receiver.excise.enabled` as well, but I'm good with this as a follow-up change.

File: pkg/cmd/roachtest/roachtestutil/operations/dependency.go (hunk 1 above)
Comment (itsbilal) [reply pkg/cmd/roachtest/roachtestutil/operations/dependency.go:32]: Thanks for catching this!

**Review by itsbilal — APPROVED** [review]
LGTM

File: pkg/cmd/roachtest/operations/cluster_settings.go (hunk 3 above)
Comment (herkolategan) [reply pkg/cmd/roachtest/operations/cluster_settings.go:92]: Yeah, that makes sense. I'll update the owner.

File: pkg/cmd/roachtest/operations/cluster_settings.go (hunk 4 above)
Comment (herkolategan) [reply pkg/cmd/roachtest/operations/cluster_settings.go:103]: Good idea, merging this one for now. Will create a follow-up PR with some additional cluster settings, I have some others in mind as well.

**Comment by herkolategan** [comment]
TFTR!

bors r=itsbilal

**Pull request merged**